### DISCUSSION TITLE
Adopt to alternative WMTS sources

### DISCUSSION THREAD

**Discussion by civilizemars** [iTowns/itowns#1639] [Q&A] [answered]
## Context

The `simple-globe` APIs gave me 403 errors and I can't get an API key.

So I wanted to test alternative WMTS sources, specifially [Mars maps from NASA](https://api.nasa.gov/?search=mars):

My question is, how I could adopt itowns to work with the `{z}/{y}/{x}` Maptile format.

`https://api.nasa.gov/mars-wmts/catalog/Mars_Viking_MDIM21_ClrMosaic_global_232m/1.0.0//default/default028mm/{z}/{y}/{x}.jpg?api_key=abcdefghi`

**Comment by gchoqueux** [comment, marked as answer]
The `api_key` url param isn't necessary. This source isn't `WTMSSource`, use `TMSSource` instead of.

```js
var marsSource = new itowns.TMSSource({
    // crs: 'EPSG:104905',
    crs: 'EPSG:4326',
    url : 'https://api.nasa.gov/mars-wmts/catalog/Mars_Viking_MDIM21_ClrMosaic_global_232m/1.0.0//default/default028mm/${z}/${y}/${x}.jpg' });

var marsLayer = new itowns.ColorLayer('wms_imagery', {
    source: marsSource,
});

view.addLayer(marsLayer);
```

![image](https://user-images.githubusercontent.com/11291849/110124202-17b81700-7dc2-11eb-867b-bbbac58aab8f.png)

With other Atmosphere parameter:

![image](https://user-images.githubusercontent.com/11291849/110124505-7382a000-7dc2-11eb-9bcc-8b8f6cbeb438.png)

**Comment by gchoqueux** [comment]
I add a this mars example in this PR  #1610

**Comment by gchoqueux** [comment]
Mars example added!
http://www.itowns-project.org/itowns/examples/#mars
